--- meson.build	2021-01-19 23:08:41.475000000 +0200
+++ meson.build	2021-01-20 00:26:27.243000000 +0200
@@ -2009,7 +2009,7 @@
   # First just find the bare library.
   libintl = cc.find_library('intl', required : false)
   # The bare library probably won't link without help if it's static.
-  if libintl.found() and not cc.has_function('ngettext', dependencies : libintl)
+  if libintl.found() and not (cc.has_function('ngettext', dependencies : libintl) or cc.has_function('ngettext', dependencies : libintl, prefix : '#include <libintl.h>'))
      libintl_iconv = cc.find_library('iconv', required : false)
      # libintl supports different threading APIs, which may not
      # require additional flags, but it defaults to using pthreads if
