--- lib/localtime-buffer.c	2019-12-21 17:35:27.182000000 -0000
+++ lib/localtime-buffer.c	2019-12-21 17:40:50.952000000 -0000
@@ -27,6 +27,12 @@
 static struct tm tm_zero_buffer;
 struct tm *localtime_buffer_addr = &tm_zero_buffer;
 
+# undef localtime
+extern struct tm *localtime (time_t const *);
+
+# undef gmtime
+extern struct tm *gmtime (time_t const *);
+
 /* This is a wrapper for localtime.
 
    On the first call, record the address of the static buffer that
