--- src/pdf2htmlEX.cc	2020-06-08 17:39:30.027000000 +0300
+++ src/pdf2htmlEX.cc	2020-06-08 18:46:17.304000000 +0300
@@ -25,6 +25,8 @@
 
 #include "pdf2htmlEX-config.h"
 
+#include "poppler/ErrorCodes.h"
+
 #if ENABLE_SVG
 #include <cairo.h>
 #endif
@@ -406,14 +408,25 @@
         }
 
         if (!doc->isOk())
-            throw "Cannot read the file";
+        {
+            int errCode = doc->getErrorCode();
+            std::cerr << "Cannot read the file. " << std::to_string(errCode) << std::endl;
+            if (errCode == errEncrypted) {
+                return 2;
+            } else {
+                return 1;
+            }
+        }
 
         // check for copy permission
         if (!doc->okToCopy())
         {
+            std::cerr << "Document has copy-protection bit set." << std::endl;
             if (param.no_drm == 0)
-                throw "Copying of text from this document is not allowed.";
-            cerr << "Document has copy-protection bit set." << endl;
+            {
+                std::cerr << "Copying of text from this document is not allowed." << std::endl;
+                return 3;
+            }
         }
 
         param.first_page = min<int>(max<int>(param.first_page, 1), doc->getNumPages());
