--- CMakeLists.txt	2020-01-26 06:00:50.339000000 +0200
+++ CMakeLists.txt	2020-01-26 06:01:56.033000000 +0200
@@ -505,6 +505,7 @@
     get_target_property(POPPLER_SOVERSION poppler SOVERSION)
     set_target_properties(poppler PROPERTIES SUFFIX "-${POPPLER_SOVERSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
 endif()
+string(STRIP "${poppler_LIBS}" poppler_LIBS)
 target_link_libraries(poppler LINK_PRIVATE ${poppler_LIBS})
 if(CMAKE_USE_PTHREADS_INIT)
    target_link_libraries(poppler LINK_PRIVATE Threads::Threads)
