--- fsys.c	2017-07-22 05:00:39.154250006 -0000
+++ fsys.c	2019-09-23 02:35:01.093000000 -0000
@@ -175,6 +175,11 @@
 return buffer;
     }
 return NULL;
+#elif defined(ANDROID)
+    static char *dir;
+    dir = getenv("HOME");
+    if ( dir!=NULL )
+	return( copy(dir) );
 #else
     static char *dir;
     uid_t uid;
@@ -893,6 +898,9 @@
 return dir;
 	}
 return NULL;
+#elif defined(ANDROID)
+	char *home = getenv("HOME");
+  return home;
 #else
 	uid_t uid;
 	struct passwd *pw;
