--- fontforge/http.c	2017-07-22 05:00:38.978162006 -0000
+++ fontforge/http.c	2019-09-26 20:20:29.102000000 -0000
@@ -90,7 +90,9 @@
 	last_len = hostent->h_length;
 	memcpy(last_addr,hostent->h_addr_list[rand()%(i+1)],hostent->h_length);
     }
+#if ! defined(ANDROID)
     endhostent();
+#endif
 return( 1 );
 }
 
@@ -254,7 +256,9 @@
 	proto = protoent->p_proto;
     else
 	proto = IPPROTO_TCP;
+#if ! defined(ANDROID)
     endprotoent();
+#endif
 
 #if MSG_NOSIGNAL!=0 || !defined(SO_NOSIGPIPE)
 return( socket(PF_INET,SOCK_STREAM,proto));
