--- gutils/fsys.c	2019-06-21 14:11:51.819753649 +0300
+++ gutils/fsys.c	2019-03-11 11:28:26.690360006 +0200
@@ -164,7 +164,7 @@
 }
 
 char *GFileGetHomeDir(void) {
-#if defined(__MINGW32__)
+#if defined(__MINGW32__) || defined(ANDROID)
     char* dir = getenv("HOME");
     if(!dir)
 	dir = getenv("USERPROFILE");
@@ -883,7 +883,7 @@
 /* reimplementation of GFileGetHomeDir, avoiding copy().  Returns NULL if home
  * directory cannot be found */
 char *getUserHomeDir(void) {
-#if defined(__MINGW32__)
+#if defined(__MINGW32__) || defined(ANDROID)
 	char* dir = getenv("APPDATA");
 	if( dir==NULL )
 	dir = getenv("USERPROFILE");
