--- CMakeLists.txt	2019-09-27 10:20:48.690000000 -0000
+++ CMakeLists.txt	2019-09-27 10:26:30.823000000 -0000
@@ -26,10 +26,9 @@
 set(PDF2HTMLEX_LIBS ${PDF2HTMLEX_LIBS} ${POPPLER_LIBRARIES})
 
 if(ENABLE_SVG)
-    pkg_check_modules(CAIRO REQUIRED cairo>=1.10.0)
+    pkg_check_modules(CAIRO-svg REQUIRED cairo-svg>=1.10.0)
     message("Trying to locate cairo-svg...")
-    find_path(CAIRO_SVG_INCLUDE_PATH cairo-svg.h PATHS ${CAIRO_INCLUDE_DIRS} NO_DEFAULT_PATH)
-    if(CAIRO_SVG_INCLUDE_PATH)
+    if(CAIRO-svg_FOUND)
         include_directories(${CAIRO_INCLUDE_DIRS})
         link_directories(${CAIRO_LIBRARY_DIRS})
         set(PDF2HTMLEX_LIBS ${PDF2HTMLEX_LIBS} ${CAIRO_LIBRARIES})
