--- fontforge/splinefont.h	2017-07-31 08:11:26.622015000 -0000
+++ fontforge/splinefont.h	2019-09-27 14:50:40.915000000 -0000
@@ -2582,9 +2582,11 @@
 
 #include "ustring.h"
 
-#ifdef _WIN32
+#if defined(_WIN32) || (defined(__ANDROID_API__) && __ANDROID_API__ < 21) 
 #define BAD_LOCALE_HACK
+#ifndef __ANDROID_API__ //Android < 21 has locale_t, but not newlocale and et cetera.
 typedef char* locale_t;
+#endif
 #define LC_GLOBAL_LOCALE ((locale_t)-1)
 #define LC_ALL_MASK LC_ALL
 #define LC_COLLATE_MASK LC_COLLATE

--- fontforge/start.c	2019-09-30 19:32:15.990000000 -0000
+++ fontforge/start.c	2019-09-30 19:35:43.297000000 -0000
@@ -47,7 +47,6 @@
 #include "psfont.h"
 
 int32 unicode_from_adobestd[256];
-struct lconv localeinfo;
 const char *coord_sep = ",";
 int quiet = 0;
 
@@ -110,10 +109,14 @@
     initadobeenc();
 
     setlocale(LC_ALL,"");
-    localeinfo = *localeconv();
     coord_sep = ",";
+
+#if (!defined(__ANDROID_API__)) || __ANDROID_API__ >= 21
+    struct lconv localeinfo = *localeconv();
     if ( *localeinfo.decimal_point=='.' ) coord_sep=",";
     else if ( *localeinfo.decimal_point!='.' ) coord_sep=" ";
+#endif
+
     if ( getenv("FF_SCRIPT_IN_LATIN1") ) use_utf8_in_script=false;
 
     inituninameannot();	/* Note: unicodenames done after locales set */

--- fontforgeexe/bitmapview.c	2017-07-22 05:00:39.054200006 -0000
+++ fontforgeexe/bitmapview.c	2019-09-30 19:36:46.154000000 -0000
@@ -46,7 +46,6 @@
 int bv_width = 270, bv_height=250;
 
 extern int _GScrollBar_Width;
-extern struct lconv localeinfo;
 extern char *coord_sep;
 struct bvshows BVShows = { 1, 1, 1, 0 };
 
--- fontforgeexe/charview.c	2019-09-30 19:29:03.035000000 -0000
+++ fontforgeexe/charview.c	2019-09-30 19:37:03.886000000 -0000
@@ -109,7 +109,6 @@
 int prefs_cv_show_control_points_always_initially = 0;
 int prefs_create_dragging_comparison_outline = 0;
 
-extern struct lconv localeinfo;
 extern char *coord_sep;
 struct cvshows CVShows = {
 	1,		/* show foreground */
